agent/api/task/task.go

```diff
@@ -984,70 +987,77 @@ func (task *Task) AddFirelensContainerBindMounts(firelensConfigType string, host
+			cniConfig.ID = eni.MacAddress
+			ifName, netconf, err = ecscni.NewBranchENINetworkConfig(eni, cniConfig)
+		default:
+			err = errors.Errorf("task config: unknown interface association type: %s",
+				eni.InterfaceAssociationProtocol)
+		}
 
-// convertENIToCNIConfig converts input eni config into cni config
-func convertENIToCNIConfig(eni *apieni.ENI, cfg *ecscni.Config) {
-	cfg.ENIID = eni.ID
-	cfg.ID = eni.MacAddress
-	cfg.ENIMACAddress = eni.MacAddress
-	cfg.SubnetGatewayIPV4Address = eni.GetSubnetGatewayIPV4Address()
-	for _, ipv4 := range eni.IPV4Addresses {
-		if ipv4.Primary {
-			cfg.ENIIPV4Address = ipv4.Address
-			break
+		if err != nil {
+			return nil, err
 		}
-	}
-	// If there is ipv6 assigned to eni then set it
-	if len(eni.IPV6Addresses) > 0 {
-		cfg.ENIIPV6Address = eni.IPV6Addresses[0].Address
+
+		cniConfig.NetworkConfigs = append(cniConfig.NetworkConfigs, &ecscni.NetworkConfig{
+			IfName:           ifName,
+			CNINetworkConfig: netconf,
+		})
 	}
 
-	// Populate Trunk ENI fields
-	if eni.InterfaceAssociationProtocol == apieni.VLANInterfaceAssociationProtocol {
-		cfg.InterfaceAssociationProtocol = eni.InterfaceAssociationProtocol
-		cfg.TrunkMACAddress = eni.InterfaceVlanProperties.TrunkInterfaceMacAddress
-		cfg.BranchVlanID = eni.InterfaceVlanProperties.VlanID
+	// Build the bridge CNI network configuration.
+	// All AWSVPC tasks have a bridge network.
+	ifName, netconf, err = ecscni.NewBridgeNetworkConfig(cniConfig, includeIPAMConfig)
+	if err != nil {
+		return nil, err
 	}
-}
+	cniConfig.NetworkConfigs = append(cniConfig.NetworkConfigs, &ecscni.NetworkConfig{
+		IfName:           ifName,
+		CNINetworkConfig: netconf,
+	})
 
-// convertAppMeshToCNIConfig converts input app mesh config into cni config
-func convertAppMeshToCNIConfig(appMesh *apiappmesh.AppMesh, cfg *ecscni.Config) {
-	cfg.AppMeshCNIEnabled = true
-	cfg.IgnoredUID = appMesh.IgnoredUID
-	cfg.IgnoredGID = appMesh.IgnoredGID
-	cfg.ProxyIngressPort = appMesh.ProxyIngressPort
-	cfg.ProxyEgressPort = appMesh.ProxyEgressPort
-	cfg.AppPorts = appMesh.AppPorts
-	cfg.EgressIgnoredIPs = appMesh.EgressIgnoredIPs
-	cfg.EgressIgnoredPorts = appMesh.EgressIgnoredPorts
+	// Build a CNI network configuration for AppMesh if enabled.
+	appMeshConfig := task.GetAppMesh()
+	if appMeshConfig != nil {
+		ifName, netconf, err = ecscni.NewAppMeshConfig(appMeshConfig, cniConfig)
+		if err != nil {
+			return nil, err
+		}
+		cniConfig.NetworkConfigs = append(cniConfig.NetworkConfigs, &ecscni.NetworkConfig{
+			IfName:           ifName,
+			CNINetworkConfig: netconf,
+		})
+	}
 
+	return cniConfig, nil
 }
 
-// isNetworkModeVPC checks if the task is configured to use task-networking feature
-func (task *Task) isNetworkModeVPC() bool {
-	if task.GetTaskENI() == nil {
-		return false
-	}
-
-	return true
+// IsNetworkModeVPC checks if the task is configured to use the AWSVPC task networking feature.
+func (task *Task) IsNetworkModeVPC() bool {
```

Comment:
Since we are changing the name of the func, is this more reasonable to be called IsAWSVPCNetworkMode?

Reply:
`IsAWSVPCNetworkMode` to me sounds like "Is AWSVPC Network Mode?", which seems like a weird name. I've renamed this to `IsNetworkModeAWSVPC` to make it more explicit.